frame/support/src/storage/types/counted_map.rs

```diff
@@ -88,6 +88,26 @@ impl<Prefix: CountedStorageMapInstance> crate::storage::PrefixIteratorOnRemoval
 	}
 }
 
+impl<Prefix, Hasher, Key, Value, QueryKind, OnEmpty, MaxValues>
+	crate::storage::generator::CountedStorageMap<Key, Value>
+	for CountedStorageMap<Prefix, Hasher, Key, Value, QueryKind, OnEmpty, MaxValues>
+where
+	Prefix: StorageInstance,
+	Hasher: crate::hash::StorageHasher,
+	Key: FullCodec,
+	Value: FullCodec,
+	QueryKind: QueryKindTrait<Value, OnEmpty>,
+	OnEmpty: Get<QueryKind::Query> + 'static,
+	MaxValues: Get<Option<u32>>,
+{
+	fn module_prefix() -> &'static [u8] {
+		Prefix::pallet_prefix().as_bytes()
+	}
+	fn storage_prefix() -> &'static [u8] {
+		Prefix::STORAGE_PREFIX.as_bytes()
+	}
+}
```

Comment:
I think we shouldn't create a new triat `CountedStorageMap` in the generator, the generator stuff were made in order for the decl_storage macro to implement its type easily.
Instead I think we should just implement the function module_prefix, storage_prefix and prefix_hash here on the type directly

Reply:
> also saying storage_prefix doesn't make sense because there is 2 storage prefix here, the one for all the values and the one for the counter, so you should probably create more function, and also call them when you injest the hashed_prefix in the tests frame/bags-list/remote-tests/src/snapshot.rs


If I understand you correctly, I should rename the `fn storage_prefix` for the `CountedStorageMap` type and also call the newly renamed `fn` in the `fn inject_hashed_prefix` in `frame/bags-list/remote-tests/src/snapshot.rs`?

Reply:
I think we should remove the file frame/support/src/storage/generator/counted_map.rs as only add some new method on the type declared in frame/support/src/storage/types/counted_map.rs
No need to create a new trait for this.

About the new method I think storage_prefix is not a good name because the counted storage map store its value under 2 prefix:
* `twox128(pallet_name) ++ twox128(storage_name)`: the values for the map
* `twox128(pallet_name) ++ twox128(CounterFor ++ storage_name)`: the counter

So maybe we can have 2 methods: `storage_map_prefix` and `storage_counter_prefix`.

Reply:
Okay, I just made some changes.